### PR TITLE
Refactor resolveUtils logic to ExtensionUtils - Feature/3.0.0 rc2

### PR DESCRIPTION
**Pull request by gracekarina**
Refactor resolveFully logic:
Now it's called addExtensions, and it adds the extensions needed at the schemas.

The responsibility of the resolveFully is now of the parser in the Class ParseOptions, always uses the resolveTrue set to true to work.

The PR looks big, because I started to change the names of the Classes from Swagger to OpenAPI.

### CODE REVIEW

File: src/main/java/io/swagger/oas/inflector/utils/ExtensionsUtil.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        if(openAPI.getPaths() != null) {
+            for (String pathname : openAPI.getPaths().keySet()) {
+                PathItem pathItem = openAPI.getPaths().get(pathname);
+                resolvePath(pathItem);
+            }
+        }
+    }
+
+     public void resolvePath(PathItem pathItem){
+        for(Operation op : pathItem.readOperations()) {
```
Comment (HugoMario) [inline src/main/java/io/swagger/oas/inflector/utils/ExtensionsUtil.java]: Please, replace "op" for a meaningful variable name. I think "op" is a brand for shoes :)

File: src/main/java/io/swagger/oas/inflector/utils/ExtensionsUtil.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+   // private static final Logger LOGGER = LoggerFactory.getLogger(ResolverUtil.class);
+
+    private Map<String, Schema> schemas;
+
+
+
+
+    public void addExtensions(OpenAPI openAPI) {
+
+        if (openAPI.getComponents().getSchemas() != null) {
```
Comment (HugoMario) [inline src/main/java/io/swagger/oas/inflector/utils/ExtensionsUtil.java]: is it save to assume that "getComponents()" will be always not null?

File: README.md [hunk 3]
```diff
@@ -41,7 +41,7 @@ To add inflector via `web.xml`:
   <servlet-class>org.glassfish.jersey.servlet.ServletContainer</servlet-class>
   <init-param>
     <param-name>javax.ws.rs.Application</param-name>
-    <param-value>io.swagger.inflector.SwaggerInflector</param-value>
+    <param-value>SwaggerInflector</param-value>
```
Comment (HugoMario) [inline README.md]: I think the class for this example is: `io.swagger.oas.inflector.OpenAPIInflector`

File: scripts/web.xml [hunk 4]
```diff
@@ -23,7 +23,7 @@
         <servlet-class>org.glassfish.jersey.servlet.ServletContainer</servlet-class>
         <init-param>
             <param-name>javax.ws.rs.Application</param-name>
-            <param-value>io.swagger.inflector.SwaggerInflector</param-value>
+            <param-value>io.swagger.oas.inflector.SwaggerInflector</param-value>
```
Comment (HugoMario) [inline scripts/web.xml]: this class does not exist on inflector new version, you should use: `io.swagger.oas.inflector.OpenAPIInflector`

**Comment by HugoMario** [comment]
@gracekarina , i added somee comments in your changes, also it would be good to add (or improve( test that confirm your changes.

File: src/main/java/io/swagger/oas/inflector/utils/ExtensionsUtil.java (hunk 2 above)
Comment (gracekarina) [inline src/main/java/io/swagger/oas/inflector/utils/ExtensionsUtil.java]: You're right. I added a validation there.

File: src/main/java/io/swagger/oas/inflector/utils/ExtensionsUtil.java (hunk 1 above)
Comment (gracekarina) [inline src/main/java/io/swagger/oas/inflector/utils/ExtensionsUtil.java]: jajaja Done!

File: README.md (hunk 3 above)
Comment (gracekarina) [inline README.md]: Done :)

File: scripts/web.xml (hunk 4 above)
Comment (gracekarina) [inline scripts/web.xml]: Fixed. 👍

File: src/main/java/io/swagger/oas/inflector/config/DefaultControllerFactory.java [hunk 5]
```diff
@@ -14,9 +14,10 @@
...
 
-package io.swagger.inflector.config;
+package io.swagger.oas.inflector.config;
 
-import io.swagger.models.Operation;
+
+import io.swagger.oas.models.Operation;
 
 /**
  * Default ControllerFactory implementation that just calls newInstance
```
Comment (HugoMario) [inline src/main/java/io/swagger/oas/inflector/config/DefaultControllerFactory.java]: what is the purpose of this class? i'm not sure if we should invoke a one line method creating an instance of this class.

**Pull request merged**